lib/sampler.py

```diff
@@ -142,11 +151,18 @@ def _sample(self):
             except ProcessLookupError:
                 self.unregister_process(pid)
             else:
-                self._sample_process(pid)
-
-    def _sample_process(self, pid):
-        # Your sampling code here.
-        pass
+                self._sample_process(pid, tasks_rss)
+        # save current overall RSS value if it is bigger than saved
+        for task in tasks_rss:
+            if self._rss_summary[task] < tasks_rss[task]:
+                self._rss_summary[task] = tasks_rss[task]
+
+    def _sample_process(self, pid, tasks_rss):
+        self.processes[pid]['rss'] = get_proc_stat_rss(pid)
```

Comment:
We don't need to store this value anywhere. We can just add it to `tasks_rss[task_id]`.

Reply:
Ok, sure, fixed in 2 places:
```
diff --git a/lib/sampler.py b/lib/sampler.py
index 11681e4..06541a4 100644
--- a/lib/sampler.py
+++ b/lib/sampler.py
@@ -106,10 +106,9 @@ class Sampler:
                 'server_name': server_name,
                 'worker_id': worker_id,
                 'worker_name': worker_name,
-                'rss': get_proc_stat_rss(pid),
             }
             task = task_id[0] + ((':' + task_id[1]) if task_id[1] else '')
-            self._rss_summary[task] = self._processes[pid]['rss']
+            self._rss_summary[task] = get_proc_stat_rss(pid)
             self._log('register', pid)
             return
 
@@ -158,11 +157,10 @@ class Sampler:
                 self._rss_summary[task] = tasks_rss[task]
 
     def _sample_process(self, pid, tasks_rss):
-        self._processes[pid]['rss'] = get_proc_stat_rss(pid)
         task_id = self._processes[pid]['task_id']
         task = task_id[0] + ((':' + task_id[1]) if task_id[1] else '')
         # count overall RSS per task
-        tasks_rss[task] = self._processes[pid]['rss'] + (tasks_rss[task] if task in tasks_rss else 0)
+        tasks_rss[task] = get_proc_stat_rss(pid) + (tasks_rss[task] if task in tasks_rss else 0)
 
 
 # The 'singleton' sampler instance: created in the main test-run
```